### PR TITLE
build: Fix format check on PRs

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Same idea as #2418 but for master and all Travis references removed.

### CODE REVIEW

**Review by mark-lunarg — APPROVED** [review]
Nate, if you're happy with this, can you add a comment to the code-format checking script (or put it in the usage/help text) that'd tell us how to use this from the command line?  For example, can I just pass in 'master' to make this work, or does it need to be HEAD~6 or a commit ID or something else?

**Comment by ncesario-lunarg** [comment]
Right now, the default usage is exactly the same as before. i.e., if you "just run the script" you will get the same behavior as before. The only difference now is that _if you want to_, you can pass any refspec to diff against (not just `origin/master`, which is the default, and I need to update the help to make it clear that is the default). Thanks to python's ArgumentParser, this is the help text that is currently generated:
```
usage: check_code_format.py [-h] [--target-refspec TARGET_REFSPEC]

Usage: python3 ./scripts/check_code_format.py
    - Reqires python3 and clang-format 7.0+
    - Run script in repo root
    - May produce inaccurate clang-format results if local branch is not rebased on the TARGET_REFSPEC
    

optional arguments:
  -h, --help            show this help message and exit
  --target-refspec TARGET_REFSPEC
                        Refspec to diff against
```
I will add something like "(default is origin/master)" to the `--target-refspec` help. Do you think this is confusing?

File: .github/workflows/ci_build.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
         sudo apt-get install -y python-pathlib
     - name: Execute Source Code Format Checking Script
       run: |
-        python3 scripts/check_code_format.py
+        python3 scripts/check_code_format.py --target-refspec=origin/${{ github.base_ref }}
```
Comment (ncesario-lunarg) [inline .github/workflows/ci_build.yml:50]: Based on the online documentation and anecdotes I found online, it sounded like `github.base_ref` was the full refspec to diff against, but it looks like that is not the case and it is just the branch name, which is why the `origin/` was needed here. Unfortunately, that probably means this still won't work for  non-master branches, but it sounds like that isn't a big issue.

**Comment by mark-lunarg** [comment]
No, sorry, I'd just missed the DEFAULT_REFSPEC part -- this is great.  LGTM!

**Pull request merged**